.github/workflows/cli-tests.yaml

```diff
@@ -95,8 +100,14 @@ jobs:
           echo "NIX_PROFILES=$NIX_PROFILES" >> $GITHUB_ENV
           echo "NIX_SSL_CERT_FILE=$NIX_SSL_CERT_FILE" >> $GITHUB_ENV
           echo "PATH=$PATH" >> $GITHUB_ENV
-      - name: Run tests
+      - name: Run go tests
         run: go test -race -cover ./...
+      - name: Run shellspec tests (bash)
+        run: shellspec testscripts/shellspec/ -s /bin/bash -f document --no-warning-as-failure
+      - name: Run shellspec tests (zsh)
+        run: shellspec testscripts/shellspec/ -s /bin/zsh -f document --no-warning-as-failure
+      - name: Run shellspec tests (dash)
+        run: shellspec testscripts/shellspec/ -s /bin/dash -f document --no-warning-as-failure
```

Comment:
These tests run on every branch now. I'm open to change it so that it only runs on main.

Reply:
I think on every branch is cool. It's much better to get feedback during PR than after merge.

Reply:
If performance of the test becomes an issue, we can split this test out and run it in parallel.